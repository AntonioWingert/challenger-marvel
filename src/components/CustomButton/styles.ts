import { styled } from "styled-components";
import { CustomButtonProps } from ".";

type ButtonProps = Omit<CustomButtonProps, 'title'>;

export const Container = styled.button<ButtonProps>`
  cursor: pointer;
  padding: ${({theme, size}) => size ? theme["button-sizes"][size] : theme["button-sizes"].md};
  border-radius: 10px;
  border: none;
  background-color: ${({theme}) => theme.colors["bg-primary"]};
  color: ${({theme}) => theme.colors["secondary"]};
  font-size: 1rem;
  font-weight: 700;
  transition: all 0.2s ease-in-out;

  &:hover {
    opacity: 0.8;
  }
`;