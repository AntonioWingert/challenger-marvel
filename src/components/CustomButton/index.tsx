import * as S from './styles';

export type CustomButtonProps = {
  title: React.ReactNode;
  size?: 'sm' | 'md' | 'lg';
} & React.ButtonHTMLAttributes<HTMLButtonElement>;

export default function CustomButton({ title, size, ...rest }: CustomButtonProps) {
  return (
    <S.Container {...rest} size={size}>
      {title}
    </S.Container>
  );
}