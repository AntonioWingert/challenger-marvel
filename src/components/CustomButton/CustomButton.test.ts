import { screen } from '@testing-library/react';
import render  from '../../utils/renderWithRouter';
import CustomButton from '.';


describe('<CustomButton />', () => {
  it('should render the button with all props passed', () => {
    const { container } = render(
      CustomButton({ title: 'Test', size: 'lg' }),
    );
    expect(screen.getByRole('button', { name: /Test/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Test/i })).toHaveStyle('padding: 2px 6px 3px 6px');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the button with default props', () => {
    render(CustomButton({ title: 'default' }));
    expect(screen.getByRole('button', { name: /default/i })).toHaveStyle('padding: 2px 6px 3px 6px');
  });
});