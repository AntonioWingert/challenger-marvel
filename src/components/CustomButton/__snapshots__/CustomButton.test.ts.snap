// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CustomButton /> > should render the button with all props passed 1`] = `
.c0 {
  cursor: pointer;
  padding: 2rem 6rem;
  border-radius: 10px;
  border: none;
  background-color: #4A1414;
  color: #F5F5F5;
  font-size: 1rem;
  font-weight: 700;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  opacity: 0.8;
}

<button
  class="c0"
>
  Test
</button>
`;
